### PR TITLE
Rework storage management

### LINKED ISSUE #43: Choose external storage as server location
Currently you can only install Cuberite in the internal storage. I don't speak for everybody, but I'm pretty sure people will likely want to have their server on their SD card due to storage limitations. I think it would be great if we were able to do so.

### PR DESCRIPTION
**Pull request by mathiascode**
We're now using the app's own external storage folder for server files. This complies with Android 10, and we no longer have to ask for write permission, which is a good idea when we run executables. NOTE: since we can no longer access the external storage, this breaks backwards compatibility with earlier versions of the app, but I think it's a necessary evil.

Support for saving server files on SD cards has been added. Fixes #43.

Crashes related to selecting files from some cloud storage providers have been fixed. An input stream is now opened directly from the file URI, instead of relying on hacky ways to get the full file path for URIs.

APK for testing: 
[app-debug.zip](https://github.com/cuberite/android/files/4021762/app-debug.zip)

### CODE REVIEW

File: app/src/main/java/org/cuberite/android/fragments/SettingsFragment.java [hunk 1]
```diff
@@ -107,6 +106,41 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
...
+                        } else if (toggleSD.isChecked()
+                                && getActivity().getExternalFilesDirs(null).length > 1) {
+                            final String SD_DIR = getActivity().getExternalFilesDirs(null)[1].getAbsolutePath();
+                            preferences.edit().putString("cuberiteLocation", SD_DIR + "/server").apply();
+                        } else {
+                            preferences.edit().putString("cuberiteLocation", PUBLIC_DIR + "/server").apply();
+                        }
+                        return true;
+                    }
+                });
```
Comment (ItsShadowCone) [inline app/src/main/java/org/cuberite/android/fragments/SettingsFragment.java:139]: I believe we should also copy / move the server dir to the new location if the user explicitly changes the location

**Review by ItsShadowCone — CHANGES_REQUESTED** [review]

**Comment by mathiascode** [comment]
I will finish this in the near future, but it's currently on the back burner due to work.

**Comment by mathiascode** [comment]
With the exception of #48, I'm postponing these changes until the storage changes in Android 11 are finalized.

**Comment by mathiascode** [comment]
Seems like Android 11 will restrict access to /Android/data, effectively killing this idea, so I'm scrapping these changes. Sticking with target API version 28 for as long as possible.

SD card support will be incorporated in another way.

**Pull request closed without merging**